packages/graphql/src/schema/validation/schema-validation.test.ts

```diff
@@ -450,75 +591,900 @@ describe("schema validation", () => {
                     }
                 `;
-
+
+                const { typeDefs: augmentedDocument } = makeAugmentedSchema(userDocument);
+                const executeValidate = () => validateUserDefinition({ userDocument, augmentedDocument });
+                expect(executeValidate).not.toThrow();
+            });
+
+            test("should not returns errors when used correctly in several place", () => {
+                const userDocument = gql`
+                    type User @authorization(filter: [{ where: { node: { id: "$jwt.sub" } } }]) {
+                        id: ID!
+                        name: String!
+                    }
+
+                    type Post @authorization(filter: [{ where: { node: { id: "$jwt.sub" } } }]) {
+                        id: ID!
+                        name: String!
+                    }
+                `;
+
+                const { typeDefs: augmentedDocument } = makeAugmentedSchema(userDocument);
+                const executeValidate = () => validateUserDefinition({ userDocument, augmentedDocument });
+                expect(executeValidate).not.toThrow();
+            });
+
+            test("should validate directive argument name", () => {
+                const userDocument = gql`
+                    type User @authorization(wrongFilter: [{ where: { node: { id: "$jwt.sub" } } }]) {
+                        id: ID!
+                        name: String!
+                    }
+                `;
+
+                const { typeDefs: augmentedDocument } = makeAugmentedSchema(userDocument);
+                const executeValidate = () => validateUserDefinition({ userDocument, augmentedDocument });
+
+                const errors = getError(executeValidate);
+                expect(errors).toHaveLength(1);
+                expect(errors[0]).not.toBeInstanceOf(NoErrorThrownError);
+                expect(errors[0]).toHaveProperty(
+                    "message",
+                    'Unknown argument "wrongFilter" on directive "@authorization". Did you mean "filter"?'
+                );
+            });
+
+            test("should validate operations value", () => {
+                const userDocument = gql`
+                    type User @authorization(filter: [{ operations: [NEVER], where: { node: { id: "$jwt.sub" } } }]) {
+                        id: ID!
+                        name: String!
+                    }
+                `;
+
+                const { typeDefs: augmentedDocument } = makeAugmentedSchema(userDocument);
+                const executeValidate = () => validateUserDefinition({ userDocument, augmentedDocument });
+
+                const errors = getError(executeValidate);
+                expect(errors).toHaveLength(1);
+                expect(errors[0]).not.toBeInstanceOf(NoErrorThrownError);
+                expect(errors[0]).toHaveProperty(
+                    "message",
+                    'Invalid argument: filter, error: Value "NEVER" does not exist in enum.'
```

Comment:
This is perhaps another example of when I maybe would have preserved the type name - up to you!